### PR TITLE
[inductor] Fix gen_transposed_tile_load_store - need_define VLA build issue on MSVC

### PR DESCRIPTION
**Pull request by xuhancn**
Recent PR: https://github.com/pytorch/pytorch/pull/131745 bring new VLA logical in cpp codegen. And it will raise build fail error on MSVC and error code is `Compiler Error C2131`: https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2131?view=msvc-170

reproduce UT:
```cmd
pytest test\inductor\test_torchinductor_dynamic_shapes.py -v -k test_large_block_sizes_dynamic_shapes_cpu
```

Original generated code:
```c++
alignas(16) float tmp1[static_cast<int64_t>(((-256LL)*(c10::div_floor_integer(static_cast<int64_t>(ks1), static_cast<int64_t>(16LL)))) + (16LL*ks1))];
```
MSVC only support C89, but VLA is need C99. We can use `std::unique_ptr` to instead of it. Reference to https://github.com/pytorch/pytorch/pull/134221

Changes:
* use unique_ptr replace vla for MSVC.
* add inductor_alloc_aligned and inductor_free for unique_ptr malloc aligned memory.
* add runtime check for inductor_alloc_aligned and inductor_free.

New genarated code:
```c++
std::unique_ptr<float, decltype(inductor_free)*> tmp1_buff((float*) inductor_alloc_aligned(sizeof(float) * static_cast<int64_t>(((-256LL)*(c10::div_floor_integer(static_cast<int64_t>(ks1), static_cast<int64_t>(16LL)))) + (16LL*ks1)), 16), inductor_free);
float* tmp1 = tmp1_buff.get();
```
Local test pass:
image

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @iremyux @Blackhex @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/codegen/cpp.py [hunk 1]
```diff
@@ -3069,11 +3069,28 @@ def gen_transposed_tile_load_store(self, name, var, index, is_store):
             tile_var = self.cse.cache[load_or_store]
 
         if need_define:
-            define_line = (
+            define_line_vla = (
                 f"alignas({factor}) {DTYPE_TO_CPP[dtype]} {tile_var}"
                 f"[{cexpr_index(self.outer_num_elems * self.inner_num_elems)}];"
```
Comment (jgong5) [inline torch/_inductor/codegen/cpp.py:3074]: Can we allocate a large-enough fixed-sized buffer here? @jiayisunx @xuhancn

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: torch/_inductor/codegen/cpp.py (hunk 1 above)
Comment (xuhancn) [reply torch/_inductor/codegen/cpp.py:3074]: Hi @jgong5 
As we discussed at perivous PR: https://github.com/pytorch/pytorch/pull/134221
1. malloc will slow down performance.
2. fixed-size buffer can't match all pattenes.
Acturally, VLA is the best performance solution. We can make MSVC work as specific compiler, and use other better compiler for performance.

Comment (xuhancn) [reply torch/_inductor/codegen/cpp.py:3074]: @jgong5 MSVC is only one compiler which not support VLA. And it is impossible to get good performance, please check the performance comparsion: https://github.com/pytorch/pytorch/pull/134772
So, let MSVC works is enough.

**Comment by xuhancn** [comment]
Will use fixed size solution: https://github.com/pytorch/pytorch/pull/135307

**Pull request closed without merging**